packages/angular_devkit/build_angular/test/browser/source-map_spec_large.ts

```diff
@@ -220,4 +221,44 @@ describe('Browser Builder source map', () => {
       }),
     ).toPromise().then(done, done.fail);
   });
+
+  it('should generate equal hash with sourcemaps on and wihout sourcemaps', (done) => {
+    const overridesWithSourceMaps: Partial<BrowserBuilderSchema> = {
+      sourceMap: true,
+      buildOptimizer: true,
+      aot: true,
+      outputHashing: OutputHashing.All,
+
+    };
+    const overridesWithoutSourcemaps: Partial<BrowserBuilderSchema> = {
+      sourceMap: false,
+      buildOptimizer: true,
+      aot: true,
+      outputHashing: OutputHashing.All,
+
+    };
+    let mainFileOutput: PathFragment | undefined;
+    let mainFileWithSourceMapOutput: PathFragment | undefined;
+
+    const runWithoutSourceMaps = runTargetSpec(host, browserTargetSpec, overridesWithoutSourcemaps)
+    .pipe(
+      tap((buildEvent) => expect(buildEvent.success).toBe(true)),
+      tap(() => {
+        mainFileWithSourceMapOutput = host.fileMatchExists(outputPath, /main\.[0-9a-f]{20}\.js/);
+      })).toPromise();
+
+
+    const runWithSourceMaps = runTargetSpec(host, browserTargetSpec, overridesWithSourceMaps).pipe(
+      tap((buildEvent) => expect(buildEvent.success).toBe(true)),
+      tap(() => {
+        mainFileOutput = host.fileMatchExists(outputPath, /main\.[0-9a-f]{20}\.js/);
+      }),
+    ).toPromise();
+
+    forkJoin(runWithoutSourceMaps, runWithSourceMaps).toPromise()
+    .then(() => {
+      expect(mainFileWithSourceMapOutput).toEqual(mainFileOutput);
```

Comment:
This test, is actually not checking that the file names are the same, but rather that both filenames contains hashes.
You should extract the hash and match it.

Reply:
Note that this is a bad test.  The file names should not be the same.  The content is different when changing the source map option which means the hash in file name should also be different.  If they were the same then this would have negative consequences for long term caching; especially when SRI is enabled (in which case the site will fail to load).

Reply:
Yes they should be different, not sure why it was marked as a bug.

Reply:
Currently it appears that they may be different for the wrong reason.  Essentially one bug partially cancelling out another.  Some further investigation may be prudent.